docs/tutorials/optimize/memory/runtime_stats.md

```diff
@@ -1,25 +1,53 @@
-### Runtime statistics
+## Runtime memory statistics
 
 Arm Mbed OS 5 provides various runtime statistics to help characterize resource usage. This allows easy identification of potential problems, such as a stack close to overflowing. The metrics currently supported are available for the [heap](#heap-statistics) and the [stack](#stack-statistics).
 
-#### Heap statistics
+### Heap statistics
 
 Heap statistics provide exact information about the number of bytes dynamically allocated by a program. It does not take into account heap fragmentation or allocation overhead. This allows allocation size reports to remain consistent, regardless of order of allocation (fragmentation) or allocation algorithm (overhead).
 
-To enable heap stats:
+#### To enable heap stats from Mbed CLI:
 
-1. Add the command-line flag `-DMBED_HEAP_STATS_ENABLED=1`.
-2. Use the function ``mbed_stats_heap_get()`` to take a snapshot of heap stats.
+1. Add the command-line flag `-DMBED_HEAP_STATS_ENABLED=1`, for example `mbed compile -DMBED_HEAP_STATS_ENABLED=1`
+1. Use the function `mbed_stats_heap_get()` to take a snapshot of heap stats.
 
-<span class="notes">**Note**: This function is available even when the heap stats are not enabled, but always returns zero for all fields.</span>
+#### To enable heap stats using `mbed_app.json`:
 
-##### Example use cases
+1. Add the following to `mbed_app.json`:
+   
+   ```
+   {
+       "macros": [
+           "MBED_HEAP_STATS_ENABLED=1"
+       ],
+       ...
+   }
+   ```
 
-* Getting worst case memory usage, `max_size`, to properly size MCU RAM.
-* Detecting program memory leaks by the current size allocated (`current_size`) or number of allocations in use (`alloc_cnt`).
-* Use `alloc_fail_cnt` to check if allocations have been failing, and if so, how many.
+1. Use the function `mbed_stats_heap_get()` to take a snapshot of heap stats.
 
-##### Example program using heap statistics
+   <span class="notes">**Note**: This function is available even when the heap stats are not enabled but always returns zero for all fields.</span>
+
+   `mbed_stats_heap_get()` returns a struct containing the following:
+
+
+   |                   |                                                   |
```

Comment:
Query: What is the empty extra row for? Can we remove it?

Reply:
Tables don't render properly without that space